src/lib/Libutil/misc_utils.c

```diff
@@ -978,17 +979,45 @@ file_exists(char *path)
 int
 is_same_host(char *host1, char *host2)
 {
-	char host1_full[PBS_MAXHOSTNAME + 1] = {'\0'};
-	char host2_full[PBS_MAXHOSTNAME + 1] = {'\0'};
+	char *host1_f = NULL;
+	char *host2_f = NULL;
+
+	static void *hostmap = NULL;
+
 	if (host1 == NULL || host2 == NULL)
 		return 0;
+
+	if (hostmap == NULL)
+		hostmap = pbs_idx_create(0, 0);
+
 	if (strcasecmp(host1, host2) == 0)
 		return 1;
-	if ((get_fullhostname(host1, host1_full, (sizeof(host1_full) - 1)) != 0)
-		|| (get_fullhostname(host2, host2_full, (sizeof(host2_full) - 1)) != 0))
+
+	pbs_idx_find(hostmap, (void **) &host1, (void **) &host1_f, NULL);
```

Comment:
I wish we had an LRU cache. With the AVL tree, we will be caching even obsolete objects.

Reply:
hostnames won't go obsolete that often right?

Reply:
Might not. This is a utility function so can be used anywhere. We don't have to rely on this assumption if the element expires after a period of time.

Reply:
I'm seeing is_same_host being invoked from svr_chk_owner.c
This might be called against any client request authorized or not. You can confirm this.

Reply:
so what? we'll cache client hosts as well, which is again good, we'll save the cost of hostnames lookups there as well, right?